packages/runtime/container-runtime/src/summarizer.ts

```diff
@@ -252,13 +261,34 @@ export class Summarizer extends EventEmitter implements ISummarizer {
         }
     }
 
-    public readonly summarizeOnDemand: ISummarizer["summarizeOnDemand"] = (...args) => {
-        if (this._disposed || this.runningSummarizer === undefined || this.runningSummarizer.disposed) {
-            throw Error("Summarizer is not running or already disposed.");
+    public readonly summarizeOnDemand: ISummarizer["summarizeOnDemand"] = async (...args) => {
+        try {
+            if (this._disposed) {
+                throw Error("Summarizer is already disposed.");
+            }
+            if (this.runningSummarizer?.disposed !== false) {
```

Comment:
I think positive check (i.e. if (this.runningSummarizer?.disposed)) is a bit more readable

Reply:
I wanted to catch the case where runningSummarizer is undefined (so the expression result is undefined) or disposed is true. I could also do that as 2 clauses.